### PR TITLE
WIP: Include ATH smoke tests into PR

### PR DESCRIPTION
**Pull request by olivergondza**
See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX).

TODO:

- [ ] ATH test results masks core ones. It seems to even hide core failures.

### Proposed changelog entries

* Entry 1: Issue, Human-readable Text
* ...

### Submitter checklist

- [ ] JIRA issue is well described
- [ ] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -69,6 +73,35 @@ for(i = 0; i < buildTypes.size(); i++) {
...
+                // Build container to run ATH
+                def uid = sh returnStdout: true, script: "id -u | tr -d '\n'"
+                def gid = sh returnStdout: true, script: "id -g | tr -d '\n'"
+                def buildArgs = "--build-arg=uid=${uid} --build-arg=gid=${gid} src/main/resources/ath-container"
+                def containerArgs = '-v /var/run/docker.sock:/var/run/docker.sock'
+                def image = docker.build('jenkins/ath', buildArgs)
+
+                unstash "jenkins.war"
+                // Run ATH inside container
+                image.inside(containerArgs) {
```
Comment (oleg-nenashev) [inline Jenkinsfile]: @jglick @abayer @svanoort  IIRC this approach is no longer recommended. amirite?

File: Jenkinsfile [hunk 2]
```diff
@@ -69,6 +73,35 @@ for(i = 0; i < buildTypes.size(); i++) {
 builds.failFast = failFast
 parallel builds
 
+stage("Acceptance Tests") {
+    node('docker && highmem') {
+        timestamps {
+            dir("ath") {
+                git url: "https://github.com/jenkinsci/acceptance-test-harness.git"
```
Comment (jglick) [inline Jenkinsfile]: Pulling from the `master` branch will make core PRs fail pretty much at random. This is unacceptable IMO.

File: Jenkinsfile (hunk 1 above)
Comment (jglick) [reply Jenkinsfile]: It is safer to `docker run` what you need, and `docker cp` out any artifacts you want the agent JVM to be able to access.

Especially for something like this that we can predict is going to be super flaky and therefore all sorts of people are going to need to know how to reproduce the problem locally. Thus the whole Pipeline script block should be:

* check out some repo
* run a designated script
* run `junit` & `archiveArtifacts`

**Review by jglick — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (olivergondza) [reply Jenkinsfile]: Sorry guys, I do not follow.

File: Jenkinsfile (hunk 2 above)
Comment (olivergondza) [reply Jenkinsfile]: So, would hardcoding a release tag we would advance every once in a while work better for you?

Comment (jglick) [reply Jenkinsfile]: Yes, a release tag would be fine. (Even a commit hash from today’s head of `master` would work, though it is less obvious in that case when updating the hash that you are going forward.)

File: Jenkinsfile (hunk 1 above)
Comment (jglick) [reply Jenkinsfile]: https://github.com/jenkinsci/docker-workflow-plugin/pull/105 for example.

**Comment by oleg-nenashev** [comment]
IIUC @raul-arabaolaza is working on a related solution

**Comment by raul-arabaolaza** [comment]
@oleg-nenashev @olivergondza Yes, related tickets are:
* [INFRA-1489](https://issues.jenkins-ci.org/browse/INFRA-1489) which has a work in progress PR [here](https://github.com/jenkins-infra/pipeline-library/pull/28) feedback welcome :)
* [JENKINS-49269](https://issues.jenkins-ci.org/browse/JENKINS-49269) Which overlaps this

**Comment by raul-arabaolaza** [comment]
I believe this can be closed now as core is already running ATH SmokeTest category due to JENKINS-49269

**Pull request closed without merging**